src/ConductorSharp.Engine/Builders/WorkflowDefinitionBuilder.cs

```diff
@@ -155,20 +147,22 @@ public void AddTask(Expression<Func<TWorkflow, DynamicForkJoinTaskModel>> reffer
                 funcase.Item2.Invoke(builder);
             }
 
-            AddTasks(builder.Build());
+            _taskBuilders.Add(builder);
+            return builder;
         }
 
-        public void AddTask<F, G>(Expression<Func<TWorkflow, JsonJqTransformTaskModel<F, G>>> refference, Expression<Func<TWorkflow, F>> input)
-            where F : IRequest<G>
-        {
-            var tasks = new JsonJqTransformTaskBuilder<F, G>(refference.Body, input.Body).Build();
-            AddTasks(tasks);
-        }
+        public ITaskOptionsBuilder AddTask<F, G>(
+            Expression<Func<TWorkflow, JsonJqTransformTaskModel<F, G>>> refference,
+            Expression<Func<TWorkflow, F>> input
+        ) where F : IRequest<G> => AddAndReturnBuilder(new JsonJqTransformTaskBuilder<F, G>(refference.Body, input.Body));
+
+        public void AddTask(Expression<Func<TWorkflow, TerminateTaskModel>> reference, Expression<Func<TWorkflow, TerminateTaskInput>> input) =>
```

Comment:
I know that it does not make sense to mark terminate task as optional but I think we still should return `ITaskOptionsBuilder` because there are other options for which we could add support sometime in the future. Also, it is possible that the optional terminate task behaves as expected and we should return this interface just for completeness

Reply:
Add it for Decision and Switch task builders